map_gen/maps/crash_site/features/vehicle_repair_beams.lua

```diff
@@ -0,0 +1,89 @@
+-- This module allows cars and tanks (any vehicle of type == car) to repair friendly structures
+local Event = require 'utils.event'
+local Global = require 'utils.global'
+
+local car_entities = {}
+
+Global.register({car_entities = car_entities}, function(tbl)
+    car_entities = tbl.car_entities
+end)
+
+Event.add(defines.events.on_entity_destroyed, function(event)
+    car_entities[event.unit_number] = nil
+end)
+
+Event.add(defines.events.on_player_driving_changed_state, function(event)
+    local entity = event.entity
+    if (entity.type ~= 'car') or not entity or not entity.valid then
```

Comment:
Probably should do the nil check before trying to access the property.
```suggestion
    if not entity or not entity.valid or (entity.type ~= 'car') then
```
Edit: Should probably do the valid check before the type check too.